projects/addon-table/components/table/directives/table.directive.ts

```diff
@@ -136,15 +135,18 @@ export class TuiTableDirective<T extends Partial<Record<keyof T, any>>>
         this.cdr.detectChanges();
     }
 
-    public updateSorter(sorter: TuiComparator<T> | null): void {
-        this.sorter = sorter || (() => 0);
-        this.sorterChange.emit(this.sorter);
-        this.change$.next();
-    }
+    public updateSorter(
+        sorter: TuiComparator<T> | null,
+        direction: TuiSortDirection = TuiSortDirection.Asc,
+    ): void {
+        if (sorter !== null && this.sorter !== sorter) {
```

Comment:
Remove all the checks. Whenever updateSorter is called, we need to set sorter, set direction and emit updates. No checks are needed.